File: data/EXP01/python_3/2257/2257.output_2.py
# @lc app=leetcode id=2257 lang=python
from typing import List

class Solution:
    def countUnguarded(self, m: int, n: int, guards: List[List[int]], walls: List[List[int]]) -> int:
        grid = [[False] * n for _ in range(m)]
        
        for guard in guards:
            row, col = guard[0], guard[1]
            for i in range(row+1, m):
                if grid[i][col]:
                    break
                grid[i][col] = True
            for i in range(row-1, -1, -1):
                if grid[i][col]:
                    break
                grid[i][col] = True
            for j in range(col+1, n):
                if grid[row][j]:
                    break
                grid[row][j] = True
            for j in range(col-1, -1, -1):
                if grid[row][j]:
                    break
                grid[row][j] = True
        
        for wall in walls:
            row, col = wall[0], wall[1]
            grid[row][col] = True
        
        count = 0
        for i in range(m):
            for j in range(n):
                if not grid[i][j]:
                    count += 1
        
        return count

File: data/EXP01/golang_no_system_var/2257/2257.output_2.go
func countUnguarded(m int, n int, guards [][]int, walls [][]int) int {
    grid := make([][]bool, m)
    for i := 0; i < m; i++ {
        grid[i] = make([]bool, n)
    }
    
    for _, wall := range walls {
        row, col := wall[0], wall[1]
        grid[row][col] = true
    }
    
    for _, guard := range guards {
        row, col := guard[0], guard[1]
        markSeenCells(grid, m, n, row, col)
    }
    
    count := 0
    for _, row := range grid {
        for _, cell := range row {
            if !cell {
                count++
            }
        }
    }
    
    return count
}

func markSeenCells(grid [][]bool, m int, n int, row int, col int) {
    directions := [][]int{{-1, 0}, {0, 1}, {1, 0}, {0, -1}}
    for _, direction := range directions {
        dx, dy := direction[0], direction[1]
        newRow, newCol := row+dx, col+dy
        
        for newRow >= 0 && newRow < m && newCol >= 0 && newCol < n && !grid[newRow][newCol] {
            grid[newRow][newCol] = true
            newRow += dx
            newCol += dy
        }
    }
}
